src/NSubstitute/Exceptions/RedundantArgumentMatcherException.cs

```diff
@@ -1,11 +1,50 @@
-﻿using System;
-using System.Collections.Generic;
+﻿using System.Collections.Generic;
 using NSubstitute.Core.Arguments;
+using static System.Environment;
 
 namespace NSubstitute.Exceptions
 {
     public class RedundantArgumentMatcherException : SubstituteException
     {
+        private static readonly string ItemSeparator = NewLine + "    ";
+
+        private static readonly string BaseMessage =
+            "Some argument specifications (e.g. Arg.Is, Arg.Any) were left over after the last call." +
+            NewLine +
+            NewLine +
+            "This is often caused by using an argument spec with a call to a member NSubstitute does not handle" +
+            " (such as a non-virtual member or a call to an instance which is not a substitute), or for a purpose" +
+            " other than specifying a call (such as using an arg spec as a return value). For example:" +
+            NewLine +
+            NewLine +
+            "    var sub = Substitute.For<SomeClass>();" + NewLine +
+            "    var realType = new MyRealType(sub);" + NewLine +
+            "    // INCORRECT, arg spec used on realType, not a substitute:" + NewLine +
+            "    realType.SomeMethod(Arg.Any<int>());" + NewLine +
```

Comment:
Have you forgot to use `Returns()` extension method there? Otherwise, the sample is very non-natural and it's totally unclear why you'd write like that 😕

Reply:
Yes, I forgot to use `.Returns()` 😆 Sorry!